### DISCUSSION TITLE
Question about PorousFlowSink

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#23566] [Q&A Modules: General] [answered]
Dear all,

I have questions regarding PorousFlowSink. 

1. I can see from the [Porous Flow Tutorial Page 06. Adding a trace](https://mooseframework.inl.gov/modules/porous_flow/tutorial_06.html) it is mentioned that it works by setting [s=f(t,x)](https://mooseframework.inl.gov/source/bcs/PorousFlowSink.html) and f as the mass rate with unit kg/m2/s.
 Now my question is what is the density referring to? component or fluid or a fluid-component mixture? If it is a mixture, can we inject only components with component density or only fluid with fluid density?  For example, in the [Porous Flow Tutorial Page 07. A chemically-reactive tracer, and porosity and permeability changes](https://mooseframework.inl.gov/modules/porous_flow/tutorial_07.html), only tracer_concentration is set by PorousFlowSink without porepressure.
```
  [BCs]
   [constant_injection_of_tracer]
     type = PorousFlowSink
     variable = tracer_concentration
     flux_function = -5E-3
     boundary = injection_area
   []
    [constant_outer_porepressure]
     type = DirichletBC
     variable = porepressure
     value = 0
     boundary = rmax
   []
 []
```

2. Compared to this PorousFlowSink, I remember when I use a point source with [Dirackernel](https://mooseframework.inl.gov/modules/porous_flow/sinks.html), it always creates some jumps with time at the source position, but this PorousFlowSink seems to give a very nice behavior without any jumps. What is the main reason resulting in that? Does PorousFLowSink only work at boundary terms?

**Comment by Joseph-0123** [comment]
Dear @GiudGiud , could you please help me have a look at this? Thank you.

**Reply by GiudGiud** [reply]
I m not really a Porous Flow expert. It would be better for @WilkAndy to weigh in.
Though I m thinking the answer to 1 will be in the tutorial or the docs of the module

**Comment by cpgr** [comment, marked as answer]
The density will be the fluid density at those conditions. If adding a component doesn't alter the density (like in the example you mentioned) then density of just that calculated for the fluid at those pressure and temperature conditions.

The sink bc is applied over the length of the boundary specified, instead of a single point like a Dirac kernel, S which probably explains the smoother change.
